src/Module/BaseApi.php

```diff
@@ -128,13 +129,7 @@ protected function addPagingConditions(array $request, array $condition): array
 				$condition = DBA::mergeConditions($condition, ["`uri-id` > ?", intval($request['min_id'])]);
 			}
 		} else {
-			switch ($requested_order) {
-				case TimelineOrderByTypes::CREATED:
-					$order_field = 'created';
-					break;
-				default:
-					$order_field = 'uri-id';
-			}
+			$order_field = $requested_order;
```

Comment:
What happens if I were to provide an arbitrary field and corresponding `max_id` from the query string? Can I have access to posts I'm not supposed to? What if the field I provide doesn't exist? Does it fail in unexpected ways?

Reply:
It is being used for sort ordering so if the field doesn't exist so the whole query will fail. For example if I accidentally set it to `created_at` instead of `created` it throws a 500 error. We could add the switch for each of the cases and have it revert to ID like I was originally intending. The column names match the enum value strings so this was more compact I thought.

Reply:
We aren't directly building the query so the sanitizer should pick that. I said I'd add the switch for all the statements so we are belaboring the point now.

Reply:
We now check for explicitly known types or throw an exception.